lib/postgrex/extensions/network.ex

```diff
@@ -0,0 +1,50 @@
+defmodule Postgrex.Extensions.Network do
+  @moduledoc false
+  alias Postgrex.TypeInfo
+  use Postgrex.BinaryExtension,
+    [send: "inet_send", send: "cidr_send", send: "macaddr_send"]
+
+  def encode(%TypeInfo{type: "inet"}, %Postgrex.INET{} = inet, _, _),
+    do: encode_network(inet)
+  def encode(%TypeInfo{type: "cidr"}, %Postgrex.CIDR{} = cidr, _, _),
+    do: encode_network(cidr)
+  def encode(%TypeInfo{type: "macaddr"}, %Postgrex.MACADDR{} = macaddr, _, _),
+    do: encode_network(macaddr)
```

Comment:
We still seem to have a few extensions that still match on `TypeInfo`. Would it be worth to split them out and should we map the oid to anonymous functions instead of modules in that case?